huaweicloud/services/waf/resource_huaweicloud_waf_dedicated_instance.go

```diff
@@ -250,36 +257,65 @@ func resourceDedicatedInstanceRead(_ context.Context, d *schema.ResourceData, me
 }
 
 // updateInstanceName call API to change the instance name.
-func updateInstanceName(c *golangsdk.ServiceClient, id, name string) error {
+func updateInstanceName(c *golangsdk.ServiceClient, id, name, epsId string) error {
 	opt := instances.UpdateInstanceOpts{
 		InstanceName: name,
 	}
-	_, err := instances.UpdateInstance(c, id, opt)
+
+	_, err := instances.UpdateWithEpsId(c, id, opt, epsId)
 	if err != nil {
 		return fmtp.Errorf("error update name of WAF dedicate instance %s: %s", id, err)
 	}
 	return nil
 }
 
 func resourceDedicatedInstanceUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	client, err := conf.WafDedicatedV1Client(conf.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("error creating HuaweiCloud WAF dedicated client: %s", err)
+	}
 	if d.HasChanges("name") {
-		config := meta.(*config.Config)
-		client, err := config.WafDedicatedV1Client(config.GetRegion(d))
+		epsId := d.Get("enterprise_project_id").(string)
+		err = updateInstanceName(client, d.Id(), d.Get("name").(string), epsId)
 		if err != nil {
-			return fmtp.DiagErrorf("error creating HuaweiCloud WAF dedicated client: %s", err)
+			return diag.FromErr(err)
 		}
-
-		err = updateInstanceName(client, d.Id(), d.Get("name").(string))
+	}
+	if d.HasChange("enterprise_project_id") && !d.IsNewResource() {
```

Comment:
Add comparison with `config.EnterpriseProjectID`, when `config.EnterpriseProjectID` is modified, the project also needs to be migrated.

Reply:
done
`epsIdChanged := d.HasChange("enterprise_project_id") ||
		d.Get("enterprise_project_id").(string) != conf.EnterpriseProjectID
	if epsIdChanged && !d.IsNewResource() {`

Reply:
ignore `config.EnterpriseProjectID`

Reply:
Ok, I see.